npm_and_yarn/lib/dependabot/npm_and_yarn/file_updater/yarn_lockfile_updater.rb

```diff
@@ -539,11 +544,16 @@ def yarnrc_specifies_npm_reg?
 
           return false unless yarnrc_global_registry
 
-          URI(yarnrc_global_registry).host == "registry.npmjs.org"
+          UpdateChecker::RegistryFinder::CENTRAL_REGISTRIES.any? do |r|
+            r.include?(URI(yarnrc_global_registry).host)
+          end
```

Comment:
Is there a better way to check for whether a config file lists private registries?
We could filter out by `CENTRAL_REGISTRIES` and check whether the output has any entries?